programs/token/src/state/mint.rs

```diff
@@ -68,13 +71,19 @@ impl Mint {
     pub unsafe fn from_account_info_unchecked(
         account_info: &AccountInfo,
     ) -> Result<&Self, ProgramError> {
-        if account_info.data_len() != Self::LEN {
+        if account_info.data_len() < Self::LEN {
             return Err(ProgramError::InvalidAccountData);
         }
-        if account_info.owner() != &ID {
+
+        if !account_info.is_owned_by(&TOKEN_2022_PROGRAM_ID)
```

Comment:
It will be more efficient for this to be specific to one token program, since a pubkey comparison consumes `12` CUs. This is another motivation to have separate crates.